File: Javascript/SubsetsII.js
var subsetsWithDuplicates = function(nums){
    const outputArray = []
    const subArray = []

    nums.sort((a,b) => a-b)

    function dfs(i){
        if(i===nums.length){
            outputArray.push([...subArray])
            return
        }

        subArray.push(nums[i])
        dfs(i+1)

        subArray.pop()
        while(i+1<nums.length && nums[i] === nums[i+1]){
            i+=1
        }
        dfs(i+1)
    }

    dfs(0)

    return outputArray
}

File: Python/SubsetsII.py
def subsetsWithDuplicates(nums):

    outputArray = []
    subArray = []
    nums.sort()

    def dfs(i):
        if i == nums.length:
            outputArray.append(subArray.copy())
            return
        
        subArray.append(nums[i])
        dfs(i+1)

        subArray.pop()
        while i+1 < len(nums) and nums[i] == nums[i+1]:
            i+=1
        dfs(i+1)

    dfs(0)
    return outputArray